Makefile

```diff
@@ -102,8 +102,8 @@ endif
 
 # Check code syntax with host compiler
 ifneq ($(RUN_CC_CHECK),0)
-  CHECK_WARNINGS := -Wall -Wextra -Wno-format-security -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-variable -Wno-missing-braces -Wno-int-conversion -Wno-unused-but-set-variable -Wno-unused-label -Wno-sign-compare -Wno-tautological-compare
```

Comment:
It does nothing currently. https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wformat-security

It warns you for cases when you misuse printf/scanf functions, like passing a float when you have a `%u`

We could annotate some variadic functions that work similar to printf (with `__attribute__ ((format (printf, 2, 3)))`) but that's a out of scope for this PR

Reply:
That's a confusing attribute. Yeah, don't think we need to worry about that.

Reply:
Forgot to link the docs about it
https://gcc.gnu.org/onlinedocs/gcc-12.2.0/gcc/Common-Function-Attributes.html#Common-Function-Attributes
I think it could actually be useful in the future